### PR TITLE
checks location input for empty search/white space

### PR DESCRIPTION
**Pull request by aboudirai**
An empty search was causing an internal server error, was able to check input before running the API request (which should help with requests/costs). Also made get_image() return None for a non-200 response code.

### CODE REVIEW

**Comment by samarmohan** [comment]
What are the conflicts?

**Comment by samarmohan** [comment]
Just staying, you should git pull, your forked repo is like 45 commits behind.

File: app.py [hunk 1]
```diff
@@ -12,25 +12,29 @@
...
+		print(type(image))
+		img_url = base64.b64encode(image.getvalue())
 
-	# print(img_url)
+		# print(img_url)
+		return render_template('index.html', image=img_url.decode('utf-8'), location=location)
 
-	return render_template('index.html', image=img_url.decode('utf-8'), location=location)
+	#GET Request Response
+	return render_template('index.html', image=False, location = False)
```
Comment (varshavjm) [inline app.py:37]: Would it make more sense to just return a simple line asking to use "POST" request instead? I am asking coz when the user enters location in this new Get page and hits submit, it still returns "Internal Server Error" which is undesirable.

**Review by varshavjm — CHANGES_REQUESTED** [review]
lgtm, I just think that the default Get request page should either point the user to use "POST" request or we should convert Internal Server error into Bad Request since "Get" semantics only retrieve data.

File: app.py [hunk 2]
```diff
@@ -12,25 +12,29 @@
 
 @app.route('/', methods=['post', 'get'])
 def login():
-	location = ''
 	if request.method == 'POST':
 		location = request.form.get('address/location')
-		img_bytes = get_image(location)
-		image = add_bernie(img_bytes)
+		if location.strip() != '':
```
Comment (varshavjm) [inline app.py:17]: This fails with AttributeError: 'NoneType' object has no attribute 'strip'. Can we also check for NoneType before the strip condition? Something like _if location is not None and location.string.strip() != '':_

Comment (aboudirai) [reply app.py:17]: looks like empty search was handled in another commit, but I'm using your suggestion to handle whitespace input! thanks!

**Comment by aboudirai** [comment]
closed since part of original fix was fixed in another commit.

**Pull request closed without merging**